ncm-download/src/main/perl/download.pm

```diff
@@ -306,12 +308,25 @@ sub retrieve
                 if (!POSIX::WIFEXITED($?) || POSIX::WEXITSTATUS($?) != 0) {
                     # Only warn as there may be attempts from multiple proxies.
                     $self->warn("curl failed (". ($?>>8) ."): $errs\nCommand = $proc");
+                    if (-e $tmpfn) {
```

Comment:
Why before? The file should not be there, but if it is, curl will truncate it. I can't think why we would want to remove it before.